docs/metrics.md

```diff
@@ -0,0 +1,652 @@
+
+## MeterProvider Management
+
+Most use-cases require you to create ONLY one instance of MeterProvider. You
+should NOT create multiple instances of MeterProvider unless you have some
+unusual requirement of having different export strategies within the same
+application. Using multiple instances of MeterProvider requires users to
+exercise caution.
+
+// TODO: Mention about creating per-thread MeterProvider // as shown in
+[this](https://github.com/open-telemetry/opentelemetry-rust/pull/2659) // PR
+
+:heavy_check_mark: Properly manage the lifecycle of `MeterProvider` instances if
+you create them. Creating a MeterProvider is typically done at application
+startup. Follow these guidelines:
+
+* **Cloning**: A `MeterProvider` is a handle to an underlying provider. Cloning
+  it creates a new handle pointing to the same provider. Clone the
+  `MeterProvider` when necessary, but re-use the cloned instead of repeatedly
+  cloning.
+
+* **Set as Global Provider**: Use `opentelemetry::global::set_meter_provider` to
+  set a clone of the `MeterProvider` as the global provider. This ensures
+  consistent usage across the application, allowing applications and libraries
+  to obtain `Meter` from the global instance.
+
+* **Shutdown**: Explicitly call `shutdown` on the `MeterProvider` at the end of
+  your application to ensure all metrics are properly flushed and exported.
+
+> [!NOTE] If you did not use `opentelemetry::global::set_meter_provider` to set
+> a clone of the `MeterProvider` as the global provider, then you should be
+> aware that dropping the last instance of `MeterProvider` implicitly calls
+> shutdown on the provider.
+
+:heavy_check_mark: Always call `shutdown` on the `MeterProvider` at the end of
+your application to ensure proper cleanup.
+
+## Memory Management
+
+In OpenTelemetry,
+[measurements](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#measurement)
+are reported via the metrics API. The SDK
+[aggregates](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#aggregation)
+metrics using certain algorithms and memory management strategies to achieve
+good performance and efficiency. Here are the rules which OpenTelemetry Rust
+follows while implementing the metrics aggregation logic:
+
+1. [**Pre-Aggregation**](#pre-aggregation): aggregation occurs within the SDK.
+2. [**Cardinality Limits**](#cardinality-limits): the aggregation logic respects
+   [cardinality
+   limits](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#cardinality-limits),
+   so the SDK does not use an indefinite amount of memory in the event of a cardinality
+   explosion.
+3. [**Memory Preallocation**](#memory-preallocation): SDK tries to pre-allocate
+   the memory it needs at each instrument creation time.
+
+### Example
+
+Let us take the following example of OTel Rust metrics being used to track the
```

Comment:
nit - to be consistent in usage of OTel or OpenTelemetry - as of now both have been used here.

Reply:
done